src/response.ts

```diff
@@ -575,29 +575,54 @@ export class Response implements DenoResponse {
     return this.send(body);
   }
 
-  // TODO: back-compat support for Express signature.
-  // Namely objects and arrays.
   /**
    * Set header `field` to `value`, or pass
    * an object of header fields.
    *
    * Examples:
    *
-   *    res.set('Accept', 'application/json');
-   *
+   *     res.set('Accept', 'application/json');
+   *     res.set({
+   *       'Accept-Language': 'en-US,en;q=0.5',
+   *       'Accept': 'text/html',
+   *     });
    * @param {string} field
    * @param {string} value
    * @return {Response} for chaining
    * @public
    */
-  set(field: string, value: string): this {
-    const lowerCaseField = field.toLowerCase();
+  set(field: string, value: string): this;
+  set(obj: Record<string, string[] | string>): this;
+  set(field: string, value: string[]): this;
+  set(field: unknown, value?: unknown): this {
+    if (arguments.length === 2) {
+      const lowerCaseField = (field + "").toLowerCase();
+      const firstVal = (Array.isArray(value) ? value[0] : value) + "";
+
+      this.headers.set(lowerCaseField, firstVal);
+
+      if (lowerCaseField === "content-type") {
+        this.type(firstVal);
+      } else {
+        this.headers.set(lowerCaseField, firstVal);
+      }
 
-    if (lowerCaseField === "content-type") {
-      return this.type(value);
-    }
+      if (Array.isArray(value)) {
+        if (lowerCaseField === "content-type") {
+          throw new TypeError("Content-Type cannot be set to an Array");
+        }
```

Comment:
This will mean that we throw in the case of `content-type` _after_ we have already set the `header` which feels a little off as a side effect, and I think we would want to avoid.

At the risk of some code duplication, I think it would be good to shift this validation check earlier.

```ts
const isArrayValue = Array.isArray(value);

if (isArrayValue && lowerCaseField === "content-type") {
  throw new TypeError("Content-Type cannot be set to an Array");
}

const firstVal = isArrayValue ? value[0] : value;
// ...remaining code
```